flume-ng-sinks/flume-dataset-sink/src/main/java/org/apache/flume/sink/kite/parser/EntityParserFactory.java

```diff
@@ -0,0 +1,82 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flume.sink.kite.parser;
+
+import java.util.Arrays;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.flume.Context;
+
+import static org.apache.flume.sink.kite.DatasetSinkConstants.*;
+
+
+public class EntityParserFactory {
+
+  public EntityParser<GenericRecord> newParser(Schema datasetSchema, Context config) {
+    EntityParser<GenericRecord> parser;
+
+    String parserType = config.getString(CONFIG_ENTITY_PARSER,
+        DEFAULT_ENTITY_PARSER);
+
+    if (parserType.equals(AVRO_ENTITY_PARSER)) {
+      parser = new AvroParser.Builder().build(datasetSchema, config);
```

Comment:
Ah. That just made it more consistent with the way the pluggable interface
works. That also makes it consistent with EventSerializers and
EventDeserializers which are related concepts.
On Tue, Jan 27, 2015 at 09:10 Ryan Blue <notifications@github.com> wrote:

> In
> flume-ng-sinks/flume-dataset-sink/src/main/java/org/apache/flume/sink/kite/parser/EntityParserFactory.java
> <https://github.com/joey/flume/pull/1#discussion_r23624064>:
>
> > +import org.apache.avro.generic.GenericRecord;
> > +import org.apache.flume.Context;
> > +
> > +import static org.apache.flume.sink.kite.DatasetSinkConstants.*;
> > +
> > +
> > +public class EntityParserFactory {
> > +
> > +  public EntityParser<GenericRecord> newParser(Schema datasetSchema, Context config) {
> > +    EntityParser<GenericRecord> parser;
> > +
> > +    String parserType = config.getString(CONFIG_ENTITY_PARSER,
> > +        DEFAULT_ENTITY_PARSER);
> > +
> > +    if (parserType.equals(AVRO_ENTITY_PARSER)) {
> > +      parser = new AvroParser.Builder().build(datasetSchema, config);
>
> Going through a builder to instantiate the AvroParser instead of doing it
> directly. Not a blocker though.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/joey/flume/pull/1/files#r23624064>.
>